### DISCUSSION TITLE
How to show breadcrumb in Search results as in SharePoint Online search

### DISCUSSION THREAD

**Discussion by vishalgoyal2472** [microsoft-search/pnp-modern-search#4094] [General]
How can we show breadcrumb as displayed in 2nd row in below image, Below image is of SharePoint online default search result

<img width="536" alt="image" src="https://github.com/user-attachments/assets/8057696b-c132-4fd5-88e1-2429882c8b7d">

**Comment by kasperbolarsen** [comment]
like this?
![image](https://github.com/user-attachments/assets/cc4df78f-98d8-43a8-a1a9-52f7220a938c)

**Reply by vishalgoyal2472** [reply]
Yes similar like yours but more details, it should be like below and similar like what we see in SharePoint online search, it should more path variables of result

`SharePoint-> Site Name-> ....->Document library or List Name`

**Reply by kasperbolarsen** [reply]
It is not too difficult, just inject this into the layout and add some css to make it look good 👍 
  <div class="template--listItem--breadcrumb">
        SharePoint > {{RefinableString00}} > {{RefinableString04}} 
          
      </div>

**Reply by vishalgoyal2472** [reply]
I have added these two columns but it is showing null in Debug template
RefinableString00
RefinableString04

**Reply by vishalgoyal2472** [reply]
I have seen below link in pnp search on github while I was comparing latest PnP solution version with my installed PnP version, does it indicate breadcrumb is already there in the latest solution and I need to upgrade to latest PnP search solution?

https://github.com/microsoft-search/pnp-modern-search/commit/e7d24f22030b148b2a4c40f0b801255a8bfcddfc

<img width="723" alt="image" src="https://github.com/user-attachments/assets/f8b89ca9-1e43-47b6-8e07-936df4cf4af5">

**Reply by kasperbolarsen** [reply]
No, not really. It is an extension component that you can drop into your custom layout, just like the snippet I mentioned earlier

<pnp-breadcrumb 
    data-path="{{OriginalPath}}"
    data-site-url="{{SPSiteURL}}"
    data-web-url="{{SPWebUrl}}"
    data-entity-title="{{Title}}"
    data-entity-file-type="{{FileType}}"
/>

**Reply by vishalgoyal2472** [reply]
you mean I need to add below section in my Template

SharePoint > {{RefinableString00}} > {{RefinableString04}}

Or I need to add below one
```
<pnp-breadcrumb 
    data-path="{{OriginalPath}}"
    data-site-url="{{SPSiteURL}}"
    data-web-url="{{SPWebUrl}}"
    data-entity-title="{{Title}}"
    data-entity-file-type="{{FileType}}"
    data-include-site-name="false" 
    data-include-entity-name="true"
    data-breadcrumb-items-as-links="true"
    data-max-displayed-items="3"
    data-overflow-index="0"
    data-font-size="12"
/>
```

**Reply by wobba** [reply]
{{RefinableString00}} > {{RefinableString04}} seems to be some custom mapping Kasper has made. So use the pnp breadcrumb component. See https://microsoft-search.github.io/pnp-modern-search/extensibility/web_components_list/#pnp-breadcrumb

**Reply by vishalgoyal2472** [reply]
I have put this pnp directly in layout but its not showing any breadcrumb, its generating html but generating same layout with all values in browser and not visible, after inspect in browser I can only see that

<img width="337" alt="image" src="https://github.com/user-attachments/assets/8c3c4ce5-b9d2-43be-a1f2-df9a2e57de85">

**Reply by vishalgoyal2472** [reply]
anyone can please respond on this?

**Reply by patrikhellgren** [reply]
Do you have OriginalPath, SPSiteURL, SPWebUrl, Title and FileType selected in Selected properties for the webpart? Also check the casing for those properties so that it is the same between Selected properties and the properties for pnp-breadcrumb.

**Reply by vishalgoyal2472** [reply]
Yes I have already added all properties and filled the properties name correctly, it does not show in search results but when I open browser in debug mode ,it shows below element to me

`<pnp-breadcrumb data-path="https://doamin-my.sharepoint.com/personal/vishal_goyal-c_com/Documents/Desktop/User Data/Profile 7/Extensions/nmmhkkegccagdldgiimedpiccmgmieda/1.0.0.6_1/craw_window.js" data-site-url="" data-web-url="[https://doamin-my.sharepoint.com/personal/vishal_goyal-c_zurichna_com"](https://doamin-my.sharepoint.com/personal/vishal_goyal-c_zurichna_com%22) data-entity-title="craw_window" data-entity-file-type="js" data-include-site-name="false" data-include-entity-name="true" data-breadcrumb-items-as-links="true" data-max-displayed-items="3" data-overflow-index="0" data-font-size="12">
</pnp-breadcrumb>`

**Reply by patrikhellgren** [reply]
In that pnp-breadcrumb element the data-site-url property is empty which tells me that either you have not selected the SPSiteURL property in Selected properties or it has an incorrect casing in your handlebars template. I have seen some tenants where it had a casing like SPSiteUrl and if it appears like that in Selected properties you have to specify it like that in your handlebars template also or else it will be empty which could be the case for you.
You could also use the debug layout view to check that all the properties I specified above has values and what casing the property names have.
The data-web-url also looks strange since it is formatted like a markdown link but maybe that could be because you copy pasted it here.

**Reply by vishalgoyal2472** [reply]
I corrected that property and below is screenshot but its still not showing in search result but showing in browser

<img width="611" alt="image" src="https://github.com/user-attachments/assets/7e99ef77-5c9a-41cd-8bee-a079b320c4e6">

**Reply by patrikhellgren** [reply]
I just tested with your first example with hard coded properties (as the breadcrumb doesn't care if we have valid urls) and with a corrected data-site-url and just below the results count in the details list layout like this:

``` html
              {{#if @root.properties.showResultsCount}}
                <label class="ms-fontWeight-semibold">{{getCountMessage @root.data.totalItemsCount @root.inputQueryText}}</label>
                <pnp-breadcrumb 
                  data-path="https://doamin-my.sharepoint.com/personal/vishal_goyal-c_com/Documents/Desktop/User Data/Profile 7/Extensions/nmmhkkegccagdldgiimedpiccmgmieda/1.0.0.6_1/craw_window.js" 
                  data-site-url="https://doamin-my.sharepoint.com/personal/vishal_goyal-c_com" 
                  data-web-url="https://doamin-my.sharepoint.com/personal/vishal_goyal-c_zurichna_com" 
                  data-entity-title="craw_window" 
                  data-entity-file-type="js" 
                  data-include-site-name="false" 
                  data-include-entity-name="true" 
                  data-breadcrumb-items-as-links="true" 
                  data-max-displayed-items="3" 
                  data-overflow-index="0" 
                  data-font-size="12"> 
                </pnp-breadcrumb>
              {{/if}}
```

After that I get this result so that is working:

![image](https://github.com/user-attachments/assets/10a273a1-f681-4dd0-8b59-a34f8ad2a150)

Since you see it in the browsers dev tools it must be hidden by css or inline styling. See if you can see a display:none in the Elements Style panel for this element or any of its parents. Can you post more of your layout since it is hard to see what is wrong by only seeing the single element.

**Reply by vishalgoyal2472** [reply]
below is the attached file for layout, thank you!!

[layout.json](https://github.com/user-attachments/files/17861695/layout.json)

**Reply by patrikhellgren** [reply]
I just tested with your layout and it works correct for me.

![image](https://github.com/user-attachments/assets/d363e980-9e40-4237-bdd9-3f148fca34aa)

What version of PnP Modern Search are you using, you didn't specify that in your initial post. It could be that you are **not** using [4.10.2](https://github.com/microsoft-search/pnp-modern-search/releases/tag/4.10.2) or later (the breadcrumb component was introduced in that version) which would give the result you are seeing.

**Reply by vishalgoyal2472** [reply]
I am using 4.6.1 version , can I upgrade my current to latest without affecting my current sites configurations which is already in use.

<img width="797" alt="image" src="https://github.com/user-attachments/assets/5d1f9da4-a37a-4da5-ac91-09172efc362f">

**Comment by kasperbolarsen** [comment]
Your current version is several versions old, but you should be able to
upgrade without issues. If you want to check it before doing it,  you can
set up an app catalog on a Site Collection,  and install the newest version
in that site collection only
